src/views/DeleteAccount.vue

```diff
@@ -0,0 +1,146 @@
+<template>
+    <b-form id="login-container">
+        <a class="close-button" @click="back"></a>
+        <div class="logo" />
+        
+        <p style="font-weight: bold">Are you sure you want to delete your account?</p>
+        <p>
+            We're sorry to hear that you're leaving us! Something wrong that we can help with?
+            Please get in touch at <a href="mailto:support@eternagame.org">support@eternagame.org</a>.
+        </p>
+        <p>
+            Your account will be deactivated and scheduled to be deleted in 30 days.
+            Per our privacy policy found at <a href="https://eternagame.org/terms">https://eternagame.org/terms</a>,
+            your identifying information will be removed and your contributions will be anonymized
+            (though we cannot retroactively remove names that may have been included in
+            scientific publications). If you have used our forum, you will also need to separately
+            delete your forum account at <a href="https://forum.eternagame.org/">https://forum.eternagame.org/</a>.
+        </p>
+        <p>
+            Thank you for joining us on our mission to advance science! To confirm you want to delete
+            your account, please enter your username and password below and click delete.
+        </p>
+        
+        <b-form-input type="text" size="md" :disabled="isLoading" v-model="username" placeholder="username"></b-form-input>
+        <b-form-input type="password" size="md" :disabled="isLoading" v-model="password" placeholder="password"></b-form-input>
+        <div class="alert-container">
+            <b-alert v-if="error" variant="danger" dismissable>
+                {{error}}
+            </b-alert>
+        </div>
+        <b-form-group>
+            <b-button variant="danger" :disabled="isLoading" size="md" @click="deleteAccount">
```

Comment:
Why is this only disabled on loading? Why not disable if username and password are empty? Rather than (or in addition to) alert.

Reply:
I suppose using errors is a guided process; if the user just clicks delete outright, they get prompted through the next steps to take.

Reply:
While the reason I didn't do it was more not considering it, for that reason I am inclined to think twice about implementing behavior like that. While forcing functions are good (don't let the user do the wrong thing in the first place), in this case there's nothing guiding the user toward what the right thing to do is - they might see the button greyed out and not know why. Good error messages at least give an indication of how to recover

Reply:
Yeah, that's a fair argument. We probably want to adjust our handling in the settings page in the future to use error messages more effectively; we require a current password to be entered to validate settings changes, but it's not immediately obvious to the end user why a password is needed before the update button is enabled. In that case, I would argue for shifting the password to a confirm modal pop up, but the point about guiding the user remains.